### PR TITLE
test(ivy): add canonical spec for object literals

### PR DESCRIPTION
**Pull request by kara**
Adds missing object literal test and corrects the test names.

Also incorporates some late feedback on 21973 from @vicb.

cc @jelbourn

### CODE REVIEW

File: packages/core/test/render3/compiler_canonical_spec.ts [hunk 1]
```diff
@@ -241,6 +242,72 @@ describe('compiler specification', () => {
...
+          static ngComponentDef = r3.defineComponent({
+            type: MyApp,
+            tag: 'my-app',
+            factory: function MyApp_Factory() { return new MyApp(); },
+            template: function MyApp_Template(ctx: MyApp, cm: boolean) {
+              if (cm) {
+                r3.E(0, ObjectComp);
+                r3.e();
+              }
+              r3.p(0, 'config', r3.o1(0, e0_literal, 'animation', ctx.name));
```
Comment (chuckjaz) [inline packages/core/test/render3/compiler_canonical_spec.ts]: Do we want `'config'` and `'animation'` to be shared if mentioned more than once?

Comment (chuckjaz) [reply packages/core/test/render3/compiler_canonical_spec.ts]: Can we get an explicit example of the nested composition such as,
```
  <object-comp [config]="{animation: name, actions: [{ opacity: 0, duration: 0}, {opacity:1, duration: duration ]"></object-comp>
```

which would probably be something like,

```
  r3.p(0, 'config', r3.o2(0, e0_literal, 'animation', ctx.name, 
     'actions': r3.o1(0, e1_literal, 1, r3.o1(0, e2_literal, 'duration', ctx.duration))));
 ...
const e0_literal = { animations: null, actions: null};
const e1_literal = [{ opacity: 0, duration: 0}, null];
const e2_literal = { opacity: 1, duration: null };
```

though I might generate the constants like,

```
const e0_literal = { animations: null, actions: null};
const c0 = { opacity: 0, duration: 0};
const e1_literal = [c0, null];
const e2_literal = { opacity: 1, duration: null };
```

**Review by chuckjaz — APPROVED** [review]

**Review by mhevery — APPROVED** [review]

File: packages/core/test/render3/compiler_canonical_spec.ts [hunk 2]
```diff
@@ -241,6 +242,152 @@ describe('compiler specification', () => {
...
+            factory: function MyApp_Factory() { return new MyApp(); },
+            template: function MyApp_Template(ctx: MyApp, cm: boolean) {
+              if (cm) {
+                r3.E(0, NestedComp);
+                r3.e();
+              }
+              r3.p(
+                  0, 'config',
+                  r3.o2(
+                      2, e0_literal_2, 'animation', ctx.name, 'actions',
```
Comment (vicb) [inline packages/core/test/render3/compiler_canonical_spec.ts]: Having keys as string literal will break once minification is enabled. 
captured in #22047

**Review by vicb — CHANGES_REQUESTED** [review]
Add minification support in that PR ?

File: packages/core/test/render3/compiler_canonical_spec.ts (hunk 2 above)
Comment (kara) [reply packages/core/test/render3/compiler_canonical_spec.ts]: This was originally a small PR to improve the tests, so it's starting to get feature creep. We can tackle that issue in a follow-up PR.

**Review by vicb — APPROVED** [review]

**Pull request closed without merging**